cmds/copy.c

```diff
@@ -127,15 +132,15 @@ static int cmd_copy(char *s)
 		return -EINVAL;
 	}
 
-	if (cmd_parseDev(&h[0], &offs[0], &sz[0], args, &argsID, argsc) < 0)
+	if (cmd_parseDev(&h[0], &offs[0], &sz[0], args, &argsID, argsc, &file[0]) < 0)
 		return -EINVAL;
 
-	if (cmd_parseDev(&h[1], &offs[1], &sz[1], args, &argsID, argsc) < 0)
+	if (cmd_parseDev(&h[1], &offs[1], &sz[1], args, &argsID, argsc, &file[1]) < 0)
 		return -EINVAL;
 
 	/* Copy data between devices */
 	log_info("\nCopying data, please wait...");
-	if (cmd_cpphfs2phfs(h[0], offs[0], sz[0], h[1], offs[1], sz[1]) < 0) {
+	if ((res = cmd_cpphfs2phfs(h[0], offs[0], sz[0], h[1], offs[1], sz[1])) < 0) {
```

Comment:
You may `return res` instead of` return -EINVAL` below when the `res` already contains a negative result, such as` EIO`, which is more descriptive than `EINVAL` in this case. This note also applies to the lines above which contain `if (cmd_parseDev (...) < 0)`, You can use `res` and return the cause of the error. What do you think about it ?

Reply:
done